platform/ui/src/components/userPreferencesForm/WindowLevelPreferences.js

```diff
@@ -1,94 +1,105 @@
-            />
-          </label>
-        </td>
-        <td className="p-r-1">
-          <label className="wrapperLabel">
-            <input
-              value={entry.window}
-              type="number"
-              className="form-control"
-              onChange={event => {
-                this.onChange(event, key, 'window');
-              }}
-            />
-          </label>
-        </td>
-        <td className="p-r-1">
-          <label className="wrapperLabel">
-            <input
-              value={entry.level}
-              type="number"
-              className="form-control"
-              onChange={event => {
-                this.onChange(event, key, 'level');
-              }}
-            />
-          </label>
-        </td>
-      </tr>
+      <td className="p-r-1">
+        <label className="wrapperLabel">
+          <input
+            value={value}
+            type={type}
+            className="form-control"
+            onChange={event => {
+              onInputChanged(event, name);
+            }}
+          />
+        </label>
+      </td>
     );
-  }
+  };
 
-  render() {
-    return (
-      <table className="full-width">
-        <thead>
-          <tr>
-            <th className="p-x-1 text-center presetIndex">Preset</th>
-            <th className="p-x-1">Description</th>
-            <th className="p-x-1">Window</th>
-            <th className="p-x-1">Level</th>
-          </tr>
-        </thead>
-        <tbody>
-          {Object.keys(this.state.data).map(key => {
-            return this.getWLPreferencesRows(key);
-          })}
-        </tbody>
-      </table>
-    );
-  }
+  return (
+    <tr key={rowName}>
+      <td className="p-r-1 text-center">{rowName}</td>
+      {renderTd(rowState.description, 'description', 'text')}
+      {renderTd(rowState.window, 'window', 'number')}
+      {renderTd(rowState.level, 'level', 'number')}
+    </tr>
+  );
 }
+/**
+ * WindowLevelPreferences tab
+ * It renders all window level presets
+ *
+ * It stores current state and whenever it changes, component messages parent of new value (through function callback)
+ * @param {object} props component props
+ * @param {string} props.name Tab`s name
+ * @param {object} props.windowLevelData Data for initial state
+ * @param {function} props.onTabStateChanged Callback function to communicate parent in case its states changes
+ */
+function WindowLevelPreferences({ windowLevelData, name, onTabStateChanged }) {
+  const [tabState, setTabState] = useState(windowLevelData);
+
+  const onWindowLevelChanged = (key, state) => {
+    setTabState({ ...tabState, [key]: state });
+  };
+
+  // tell parent to update its state
+  useEffect(() => {
+    onTabStateChanged(name, { windowLevelData: tabState });
+  }, [tabState]);
+
+  return (
+    <table className="full-width">
+      <thead>
+        <tr>
+          <th className="p-x-1 text-center presetIndex">Preset</th>
+          <th className="p-x-1">Description</th>
```

Comment:
It's worth adding translations for this component? maybe its worth backlog a note for this.

Reply:
gr8, we will need to revisit it anyway for windowLevel impl... once that time comes we could cover this part

Reply:
Can one of you create a community issue for this? Tasks like these can be great on-boarding tasks for community members or new Radical Product Team members as they onboard.

First one to create the issue gets a virtual cookie 🍪

Reply:
> Can one of you create a community issue for this? Tasks like these can be great on-boarding tasks for community members or new Radical Product Team members as they onboard.
> 
> First one to create the issue gets a virtual cookie 🍪

https://github.com/OHIF/Viewers/issues/1209

Reply:
@dannyrb wheres my cookie?

Reply:
https://media2.giphy.com/media/bAlYQOugzX9sY/giphy.gif?cid=790b7611860aad18ad356e05f31345d3f4e5d95916a13310&rid=giphy.gif